site/cypress/support/commands.js

```diff
@@ -24,50 +24,71 @@
 // Cypress.Commands.overwrite("visit", (originalFn, url, options) => { ... })
 
 import 'cypress-file-upload';
-import {buildUrl} from './utils.js';
+import { buildUrl } from './utils.js';
 //These functions can be called like "cy.login(...)" and will yield a result
 
 /**
-* Log into Submitty, assumes no one is logged in already and at login page
+* Log into Submitty using API, if a user is already logged in, you are redirected to 
+* the My Courses page (/home)
 *
 * @param {String} [username=instructor] - username & password of who to log in as
 */
-Cypress.Commands.add('login', (username='instructor') => {
+Cypress.Commands.add('login', (username = 'instructor') => {
+    cy.url({ decode: true }).then($url => {
+        cy.request({
+            method: 'POST',
+            url: '/authentication/check_login'.concat('?', $url.split('?')[1]),
+            form: true,
+            followRedirect: false,
+            body: {
+                user_id: username,
+                password: username,
+                __csrf: username,
+            },
+        }).then(response => {
+            cy.visit(response.redirectedToUrl);
+        })
+    })
+});
+
+/**
+* Log into Submitty using the UI, assumes no one is logged in already and at login page
+*
+* @param {String} [username=instructor] - username & password of who to log in as
+*/
+Cypress.Commands.add('loginViaUI', (username = 'instructor') => {
```

Comment:
Why does this need to be a command?  Are there any cases where we might want to login by clicking around the UI instead of just making a HTTP request directly?  Shouldn't this just be a test in the login tests?